fluid/PaddleCV/metric_learning/imgtool.py

```diff
@@ -35,14 +42,18 @@ def random_crop(img, size, scale=None, ratio=None):
     scale_max = min(scale[1], bound)
     scale_min = min(scale[0], bound)
 
-    target_area = img.shape[0] * img.shape[1] * random.uniform(scale_min,
-                                                             scale_max)
+    target_area = img.shape[0] * img.shape[1] * random.uniform(scale_min, scale_max)
+    if is_ce:
+        target_area = img.shape[0] * img.shape[1] * (scale_min + scale_max) / 2.
     target_size = math.sqrt(target_area)
     w = int(target_size * w)
     h = int(target_size * h)
 
     i = random.randint(0, img.shape[0] - h)
     j = random.randint(0, img.shape[1] - w)
+    if is_ce:
+        i = int(img.shape[0] - h) // 2
+        j = int(img.shape[1] - w) // 2
```

Comment:
You also can set rand seed for numpy, please refer image classification. So do not add `ce` branch in reader.